### PR TITLE
Fix session tracker backwards compatibility.

### PR DESCRIPTION
**Pull request by Joerger**
Forward port of https://github.com/gravitational/teleport/pull/12728

This isn't strictly necessary, since v10 auth servers aren't expected to be fully compatible with v9 services (nodes, etc.), but it's an easy forward port and was requested.

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

File: lib/srv/app/session.go [hunk 1]
```diff
@@ -261,6 +261,14 @@ func (s *Server) createTracker(sess *sessionChunk, identity *tlsca.Identity) err
 	s.log.Debugf("Creating tracker for session chunk %v", sess.id)
 	tracker, err := srv.NewSessionTracker(s.closeContext, trackerSpec, s.c.AuthClient)
 	if err != nil {
+		// Ignore access denied errors, which we will get if the auth
+		// server is v9.2.1 or earlier, since only node, proxy, and
+		// kube roles had permission to create session trackers.
+		// DELETE IN 11.0.0
+		if trace.IsAccessDenied(err) {
```
Comment (codingllama) [inline lib/srv/app/session.go:268]: Suggestion: Same as above, use a switch to remove nesting?

Ditto for others.

File: lib/srv/db/server.go [hunk 2]
```diff
@@ -899,6 +899,14 @@ func (s *Server) trackSession(ctx context.Context, sessionCtx *common.Session) e
 	s.log.Debugf("Creating tracker for session %v", sessionCtx.ID)
 	tracker, err := srv.NewSessionTracker(s.closeContext, trackerSpec, s.cfg.AuthClient)
 	if err != nil {
+		// Ignore access denied errors, which we will get if the auth
+		// server is v9.2.1 or earlier, since only node, proxy, and
+		// kube roles had permission to create session trackers.
+		// DELETE IN 11.0.0
+		if trace.IsAccessDenied(err) {
```
Comment (codingllama) [inline lib/srv/db/server.go:906]: Suggestion: Same as above, use a switch to remove nesting?

**Review by codingllama — COMMENTED** [review]
I'm not super familiar with event upload logic, so it's a bit hard to reason about the fact that all AccessDenied errors are safe to swallow.

Is it worth trying to put a test together for this situation?

What do we get by forward porting if it's not necessary? Could this have undesirable side-effects?

**Comment by Joerger** [comment]
> I'm not super familiar with event upload logic, so it's a bit hard to reason about the fact that all AccessDenied errors are safe to swallow.

We expect to get `NotFound` from v8 and earlier v9 versions, and have the grace period backwards compatibility logic to handle this case - https://github.com/gravitational/teleport/blob/b780f0de27898301f0704cd6fd896f5f8b992a44/lib/events/complete.go#L158-L161
With this change we are essentially handling `AccessDenied` errors in the same way, and defaulting to the grace period logic. So even if we got `AccessDenied` for another reason (although we never should), then we still have the grace period to fall back on.

> Is it worth trying to put a test together for this situation?

Sure, I've added it to the grace period test.

> What do we get by forward porting if it's not necessary? Could this have undesirable side-effects?

Although we don't formally support `v9` auth with `v10` db/app/desktop services, users still do it frequently enough that it would be better from a support standpoint handle fatal errors like this one, which returns an error for any session started with these services.

File: lib/events/complete_test.go [hunk 3]
```diff
@@ -248,3 +257,29 @@ func (m *mockSessionTrackerService) RemoveSessionTracker(ctx context.Context, se
 func (m *mockSessionTrackerService) UpdatePresence(ctx context.Context, sessionID, user string) error {
 	return trace.NotImplemented("UpdatePresence is not implemented")
 }
+
+type mockSessionTrackerServiceAccessDenied struct{}
+
+func (m *mockSessionTrackerServiceAccessDenied) GetActiveSessionTrackers(ctx context.Context) ([]types.SessionTracker, error) {
+	return nil, trace.AccessDenied("Access denied")
```
Comment (codingllama) [inline lib/events/complete_test.go:264]: super nit: uppercase error string

```suggestion
	return nil, trace.AccessDenied("access denied")
```

Same for others.

**Review by codingllama — APPROVED** [review]
Thanks for the explanation, Brian. Very helpful.

**Pull request merged**